### DISCUSSION TITLE
Does  zitadel need HTTP2

### DISCUSSION THREAD

**Discussion by blueray453** [zitadel/zitadel#4106] [Q&A] [answered]
In https://github.com/zitadel/zitadel#client-libraries , it talks about grpc

HTTP 2.0 is required for gRPC

Does zitadel require HTTP 2.0?

For which features zitadel require HTTP 2.0?

Does it work with REST APIs.

**Comment by fforootd** [comment, marked as answer]
Hi There

ZITADEL makes use of [HTTP/2 for different things](https://docs.zitadel.com/docs/guides/manage/self-hosted/http2).

- Faster Loading of assets
- Faster API responses when using gRPC-web and REST

If you deploy ZITADEL you only need to make sure that the connection that directly interacts with your Instance can do HTTP/2.
So if you deploy a proxy in front of ZITADEL please make sure that HTTP/2 works towards ZITADEL.

On the Front your proxy can use HTTP/1.1 towards your clients, as long as you do not use gRPC.
Our REST service and as well the gRPC-web API is fine with HTTP/1.1.

Each API is available in gRPC, gRRPC-web and REST. [You can find more details here.](https://docs.zitadel.com/docs/apis/introduction)

Hope this helps

**Reply by blueray453** [reply]
Thank you very much.

**Reply by fforootd** [reply]
Happy to help.

If this satisfies your questions please close this as answered 😁

**Reply by blueray453** [reply]
I can not find the close button.

**Reply by fforootd** [reply]
Am sorry that is only available in QA 😁 

I just moved this discussion. Now there should be an "Mark as answer" button.
